src/pages/Admin/Room/Edit/RoomEditModal.tsx

```diff
@@ -0,0 +1,194 @@
+    undefined
+  );
+
+  useEffect(() => {
+    const template = ROOM_TEMPLATES.find(
+      (roomTemplate) => roomTemplate.template === selectedTemplate
+    );
+    setRoomTemplate(template);
+  }, [selectedTemplate]);
+
+  const {
+    register,
+    watch,
+    handleSubmit,
+    setValue,
+    errors,
+    formState: { isSubmitting, dirty },
+  } = useForm({
+    validationSchema: roomEditSchema,
+    defaultValues: {
+      title: room.title,
+      url: room.url,
+      description: room.description,
+      template: room.template,
+      image_url: room.image_url,
+    },
+  });
+
+  const values = watch();
+
+  const handleImageChange = useCallback(
+    (val: string) => {
+      setValue("image_url", val, false);
+    },
+    [setValue]
+  );
+
+  const onSubmit = useCallback(() => {
+    return submitHandler(values, room.roomIndex!);
+  }, [room.roomIndex, submitHandler, values]);
+
+  const customInputFields = useMemo(
+    () =>
+      roomTemplate?.customInputs?.map((input: CustomInputsType) => (
+        <div className="room-edit-modal__input" key={input.name}>
+          <span>{input.title}</span>
+          <input type="text" name={input.name} ref={register} />
+        </div>
+      )),
+    [register, roomTemplate]
+  );
+
+  return (
+    <Modal
+      show={isVisible}
+      onHide={onClickOutsideHandler}
+      className="room-edit-modal"
+    >
+      <Modal.Header>
+        <Modal.Title>Editing - {room.title}</Modal.Title>
+      </Modal.Header>
+
+      <Modal.Body>
+        <Form onSubmit={handleSubmit(onSubmit)}>
+          <Form.Row>
+            <div className="room-edit-modal__input">
+              <Form.Label>Name your room</Form.Label>
+              <Form.Control
+                type="text"
+                ref={register}
+                name="title"
+                placeholder="Room name"
+                custom
+              />
+              {errors.title && (
+                <span className="input-error">{errors.title.message}</span>
+              )}
+            </div>
+          </Form.Row>
+
+          <Form.Row>
+            <div className="room-edit-modal__input">
+              <Form.Label>The room url</Form.Label>
+              <Form.Control
+                type="text"
+                ref={register}
+                name="url"
+                placeholder="Room url"
+                custom
+              />
+              {errors.url && (
+                <span className="input-error">{errors.url.message}</span>
+              )}
+            </div>
+          </Form.Row>
+
+          <Form.Row>
+            <div className="room-edit-modal__input">
+              <Form.Label>The room description (optional)</Form.Label>
+              <Form.Control
+                type="text"
+                ref={register}
+                name="description"
+                placeholder="Description"
+                custom
+              />
+            </div>
+          </Form.Row>
```

Comment:
Should this row also have an error section on it? (my thoughts are probably yes, even if we don't currently have an explicit validation setting an error on it, having that consistent 'wiring' in place makes sense to me)

Reply:
In the staging `RoomEdit.tsx` file, it wasn't there so I didn't add it, but I will now

Reply:
I added an error section but at the moment it shouldn't validate anything (as there isn't really anything a description shouldn't be; maybe a maximum length could be implemented?). In my opinion, adding the error validation for this is OOS for this PR.

Reply:
Agreed that the validation itself is OOS, just wanted to make sure the 'slots' were in place for it.